src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -178,6 +178,17 @@ trait Infer extends Checkable {
     def context: Context
     import InferErrorGen._
 
+    // flag used to turn off tupling when alternatives of an overload include a Java-defined method
+    private[this] var tupleForApplicability = true
+    def withoutTuplingForApplicability[A](body: => A): A =
```

Comment:
In fact, I was prompted to look at this PR this afternoon, and didn't like the vars. I always get an "I wrote that?" feeling, but this wasn't pleasant. I couldn't see init order and usages at my accustomed glance; maybe I can find the guy who wrote it and get him to work it up a bit.